library/clinical_rules.php

```diff
@@ -1025,85 +1025,151 @@ function test_rules_clinic($provider = '', $type = '', $dateTarget = '', $mode =
+
+                        test_targets considers all targets as "inclusion" even if target is defined as "exclusion"
+
+                        I added $dateTarget param to call to test_targets, to allow right boundary of examined intervals to be $dateTarget regardless of $dateFocus value
+                        */
+                        $passTarget = test_targets($rowPatient['pid'], $rowRule['id'], '', $dateFocus, $dateTarget);
+
+                        if ($passTarget) {
+                            // increment pass target counter
+                            $pass_target++;
+                            // If report itemization is turned on, then record the "passed" item and set the flag
+                            if (!empty($GLOBALS['report_itemizing_temp_flag_and_id'])) {
+                                insertItemReportTracker($GLOBALS['report_itemizing_temp_flag_and_id'], $GLOBALS['report_itemized_test_id_iterator'], 1, $rowPatient['pid']);
+                                $temp_track_pass = 1;
                             }
-                        }
 
-                        break;
-                    } else {
-                        // send to reminder results
-                        if ($mode != "report") {
-                            // place the uncompleted actions into the reminder return array
-                            $actionArray = resolve_action_sql($rowRule['id'], '1');
-                            foreach ($actionArray as $action) {
-                                $action_plus = $action;
-                                $action_plus['due_status'] = $reminder_due;
-                                $action_plus['pid'] = $rowPatient['pid'];
-                                $action_plus['rule_id'] = $rowRule['id'];
-                                $results = reminder_results_integrate($results, $action_plus);
+                            // send to reminder results
+                            if ($mode == "reminders-all") {
+                                // place the completed actions into the reminder return array
+                                $actionArray = resolve_action_sql($rowRule['id'], '1');
+                                foreach ($actionArray as $action) {
+                                    $action_plus = $action;
+
+                                    // original line
+                                    // With this line, Reminder Details page shows only not due and past due
+                                    //$action_plus['due_status'] = "not_due";
+
+                                    // HR: My replacement lines
+                                    // if passed during this pass, this is the status
+                                    if ($dateCounter == 1) {
+                                        // not_due doesn't cause any text to show. Labels come from Clinical Rules Reminder Due Options. Has not_due, so not sure why that does not work
```

Comment:
note to self:
look into this specifically on testing

Reply:
posted fix for the not_due label in the core codebase at:
https://github.com/openemr/openemr/pull/7201